packages/zowe-explorer-api/src/fs/utils/datasets.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { IFileSystemEntry, DsEntry, MemberEntry, PdsEntry } from "../";
+import { FileType } from "vscode";
+
+export function isDsEntry(entry: IFileSystemEntry): entry is DsEntry {
+    return entry != null && entry.type == FileType.File;
```

Comment:
I may be reading this wrong, but wouldn't this return true for a USS file entry too?

Reply:
Yep, you're right. I changed it to use `instanceof` since `DsEntry` is a class anyway. I think this was from a while back when the foundation was still an experiment of sorts.